test/api_integration/apis/status/status.js

```diff
@@ -0,0 +1,47 @@
+import expect from 'expect.js';
+
+export default function ({ getService }) {
+  const supertest = getService('supertest');
+
+  describe('kibana status api', () => {
+    it('returns version, status and metrics fields', () => {
+      return supertest
+        .get('/api/status')
+        .expect('Content-Type', /json/)
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.name).to.be.a('string');
+          expect(body.uuid).to.be.a('string');
+          expect(body.version.number).to.be.a('string');
+          expect(body.version.build_hash).to.be.a('string');
+          expect(body.version.build_number).to.be.a('number');
+
+          expect(body.status.overall).to.be.an('object');
+          expect(body.status.overall.state).to.be('green');
+
+          expect(body.status.statuses).to.be.an('array');
+          const kibanaPlugin = body.status.statuses.find(s => {
+            return s.id.indexOf('plugin:kibana') === 0;
+          });
+          expect(kibanaPlugin.state).to.be('green');
+
+          expect(body.metrics.collection_interval_in_millis).to.be.a('number');
+
+          expect(body.metrics.process.mem.heap_max_in_bytes).to.be.a('number');
+          expect(body.metrics.process.mem.heap_used_in_bytes).to.be.a('number');
+
+          expect(body.metrics.os.cpu.load_average['1m']).to.be.a('number');
```

Comment:
If we were to run this on Windows, I guess we currently return `0`, so we can safely do this and fix the test when we fix that.

Reply:
Hi, sorry but I'm not following what fix you had in mind. If we're in windows, then should it give an undefined for these fields?

Reply:
Exactly. But nothing for this PR. We currently return `0` for Windows, but we should be returning `undefined`. We can do that explicitly in a follow-on PR.

Reply:
https://github.com/elastic/kibana/issues/17548